src/style.css

```diff
@@ -0,0 +1,95 @@
+* {
+  box-sizing: border-box;
+}
+
+html {
+  font-family: Roboto, sans-serif;
+  font-weight: 400;
+  font-size: 11px;
+}
+
+body,
+p {
+  margin: 0;
+  padding: 0;
+}
+
+.main__card {
+  width: 200px;
+  padding: 14px;
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+}
+
+.product-image {
+  height: 134px;
+  width: 160px;
+  margin: 16px 5px 45px;
+  display: flex;
+  justify-content: center;
+}
+
+.main__product-name {
+  font-size: 12px;
+  font-weight: 500;
+  line-height: 15px;
+  margin-bottom: 5px;
+}
+
+.main__product-code {
+  margin-bottom: 15px;
+}
+
+.main__rating-container {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 20px;
+}
+
+.main__product-stars {
+  display: flex;
+  justify-content: space-between;
+  flex-direction: row;
+  align-items: center;
+  height: 15px;
+  margin-left: -4px;
+}
+
+.main__product-star {
+  width: 15px;
+  height: 12px;
+  margin-left: 4px;
+}
+
+.main__product-price {
+  display: flex;
+  justify-content: space-between;
+  align-items: baseline;
+  margin-bottom: 20px;
+}
+
+.main__product-price-value {
+  font-size: 16px;
+  font-weight: 700;
+}
+
+.main__button {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 40px;
+  font-weight: 700;
+  font-size: 14px;
+  text-align: center;
+  text-decoration: none;
+  color: #fff;
+  background-color: #00acdc;
+  border-radius: 5px;
+}
+
+.main__button:hover {
+  color: #00acdc;
+  background-color: #fff;
+  border: 1px solid #00acdc;
```

Comment:
at this moment when you hovered on your link - the border appears and adds 1px, because of box-sizing: border-box; width and height of your button is recalculating, and text could change its position, so it's better if you write border property inside the button.